### ISSUE TITLE
LOAD httpfs; creates fatal error on Windows (from R)

### ISSUE THREAD

**Issue by cboettig** [duckdb/duckdb#6231]
### What happens?

The following sequence of commands crashes R (or RStudio) with fatal error and no output. Other platforms seem to work fine. 

### To Reproduce

```r
conn <- DBI::dbConnect(duckdb::duckdb())
DBI::dbExecute(conn, "INSTALL 'httpfs';")
DBI::dbExecute(conn, "LOAD 'httpfs';")
```

### OS:

Windows 10

### DuckDB Version:

latest

### DuckDB Client:

R

### Full Name:

Carl Boettiger

### Affiliation:

UC Berkeley

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mause** [comment]
Are you using the latest release version or the latest prerelease version?

**Comment by cboettig** [comment]
latest release on CRAN

**Comment by cboettig** [comment]
(for context: I normally develop on linux and maintain several R packages on CRAN that themselves use duckdb. this issue prevents those packages from working for any Windows users if I try and include operations that take advantage of httpfs extension.  Linux and Mac seem not to be impacted.  I have a cheap Windows 10 tablet where I can confirm this bug, but can't build duckdb from source on that Windows platform so couldn't test the pre-release).

**Comment by Pascallio** [comment]
I have this exact issue. Rstudio produces a fatal error when loading the httpfs extension on Windows.

**Comment by Mytherin** [comment]
Thanks for the report! I have looked into this and it seems to be caused by R using the mingw toolchain on Windows, which does  not seem to be compatible with standard Windows C++ libraries compiled by MSVC. To fix this we will likely need to compile a different set of extensions using the MinGW architecture.

**Comment by cboettig** [comment]
Thanks @Mytherin! any news on this?  would it be possible to intercept the execution on windows from R meanwhile so users at least get a good error message instead of crashing out?

**Comment by Mytherin** [comment]
Yes, that's a great idea and a relatively simple fix. I have pushed that fix in #7007. The actual work in setting up CI builds for building, uploading and testing all extensions on MinGW is significantly more work so that will have to wait until someone has spare time to pick it up.
